app/controllers/api/series_controller.rb

```diff
@@ -47,6 +47,26 @@ def create_resource
     end
   end
 
+  # XXX this method is copied from the ResourceCallbacks concern
+  def create
+    json_body_params = JSON.parse(request.body)
+
+    series = create_resource
+    consume! series, create_options
+    hal_authorize series
+
+    @series = if json_body_params.has_key?('import_url')
```

Comment:
It's possible. I think the problem lies in the `create_from_feed` call below.  As I understand it, roar/hal_api_rails must create model instances in a rails conventional manner, so a custom method to create the series from a feed mucks that up.

Another tact to clean this up would be to expose these params via an internal API (e.g. `hal_params.permit(:import_url)`.  But these params are not exposed anywhere (?), which in hindsight appears to be a conscious decision in the hal_api/roar/representer world.

Assuming we stick with something like what's currently here, these params likely need to be wrapped in strong params to prevent nested attr hacks.

Reply:
Ah, I see.  So you'd need to somehow get the `current_user` and `account` at the same time.  And even then ... complicated.  If there has to be some "magic" somewhere, I'd prefer it here in the controller.

Does this properly belong here in the non-auth series controller?

Reply:
Re: `Does this properly belong here in the non-auth series controller?`

My instinct is to say no. But, there is some protection here, up on line 42 we check for the `current_user` which is enough to reject anonymous access (with a 500--access attr on nil).

It looks like we are missing a corresponding set of account scoped series routes within the api auth scope.  If we create those new routes (at least for the create method here), we'd get proper 4xx semantics. Worth it?

Reply:
Yeah, let's track that separately.  In theory, those policy classes should be restricting write access to resources, and returning a 4xx long before it gets there.  May just be an edge case with the use of current_user.

So we should hash this out at some point ... but as we get more into this with ID creating accounts, my feelings are that it's fine to PUT/POST/DELETE to public api endpoints.  The `/api/v1/authorizations` routes were created to solve (1) caching issues on GET, and (2) filtering resources based on an auth token.  And neither of those are a consideration if you're creating/updating/deleting things.